### PR TITLE
Filter output by problem level

### PR DESCRIPTION
**Pull request by ffapitalle**
Add configuration option `output-verbosity` to enable output filtering by problem level.

Aims to solve #86.

### CODE REVIEW

**Comment by adrienverge** [comment]
Hello @ffapitalle,

Can you explain the use case (indenpendently from #86)?

If the goal is to suppress warnings *using conf*, why not using `level: error` or `disable` *in conf*?

**Comment by ffapitalle** [comment]
Hi @adrienverge,

Main goal is to suppress from the output warning level problems and show only errors.

My use case is exactly the same as #86, I've a proyect with a lot of files that are being validated automatically, and as a result I want a notification containing only errors. Rules marked as warnings should keep reporting as warnings.

I can use also a command line argument, like `-q` or `--no-warnings`, but as I found the former issue , I went that way.

**Comment by adrienverge** [comment]
OK thanks, I understand the use case.

I find confusing and error-prone to have a conf that contradicts itself (*enable this rule as warning*, but *finally don't report it*).

A command line argument makes more sense, as it's temporary and explicitly passed when running the command. Also, an argument would be coherent with `-s` / `--strict`. Would you be OK to change your pull request to use `--no-warnings`?

**Comment by ffapitalle** [comment]
> Would you be OK to change your pull request to use `--no-warnings`?

Agreed. I've have already another branch that adds a command line argument. Would you prefer open a new one ? Or add the changes to this request ?

**Comment by adrienverge** [comment]
As you prefer!

**Comment by ffapitalle** [comment]
> Add configuration option `output-verbosity` to enable output filtering by problem level.
> 
> Aims to solve #86.

Changed proposed solution: instead of adding a configuration option, I've added a command line option, which supress printing of warning level problems on script output.

File: docs/configuration.rst [hunk 1]
```diff
@@ -116,6 +116,9 @@ return code will be:
  * ``1`` if one or more errors occur
  * ``2`` if no errors occur, but one or more warnings occur
 
+If the script is invoked with the ``-q`` (or ``--no-warning``) options, it will
```
Comment (adrienverge) [inline docs/configuration.rst]: - The commit names and previous comments talk about `--no-warnings` (with an `s`), I think it's better English (although I'm not a native English speaker) and more coherent with other options.

- I'd prefer reserve `-q` for later use, moreover the letter `q` would match `--quiet`, but no direct link with `--no-warning`. Can you use only `--no-warnings`? (Same for variable names `quiet` in other files)

File: tests/test_cli.py [hunk 2]
```diff
@@ -532,3 +532,29 @@ def test_run_read_from_stdin(self):
             'stdin:2:10: [error] syntax error: '
             'mapping values are not allowed here\n'))
         self.assertEqual(err, '')
+
+    def test_run_supress_warnings(self):
```
Comment (adrienverge) [inline tests/test_cli.py]: `supress` → `suppress` 

or `test_run_no_warnings()`?

File: yamllint/cli.py [hunk 3]
```diff
@@ -84,11 +84,14 @@ def standard_color(problem, filename):
...
 
 
-def show_problems(problems, file, args_format):
+def show_problems(problems, file, args_format, quiet):
     max_level = 0
     first = True
 
     for problem in problems:
+        max_level = max(max_level, PROBLEM_LEVELS[problem.level])
+        if quiet and (not problem.level == 'error'):
```
Comment (adrienverge) [inline yamllint/cli.py]:
```suggestion
        if quiet and problem.level != 'error':
```

File: yamllint/cli.py [hunk 4]
```diff
@@ -84,11 +84,14 @@ def standard_color(problem, filename):
         return line
 
 
-def show_problems(problems, file, args_format):
+def show_problems(problems, file, args_format, quiet):
     max_level = 0
     first = True
 
     for problem in problems:
+        max_level = max(max_level, PROBLEM_LEVELS[problem.level])
```
Comment (adrienverge) [inline yamllint/cli.py]: I'm curious: why did you move this line?

Is it to keep a non-zero return code if `--strict` is used in the same time as `--no-warnings`? (I think it's a good thing). Can you add a `test_run_no_warnings_and_strict()`? for that?

File: docs/configuration.rst (hunk 1 above)
Comment (ffapitalle) [reply docs/configuration.rst]: You're right, I've missed the `s` on the doc file, whereas everywhere else I've used `warnings`.

I've also removed `-q` and kept only `--no-warnings`.

File: tests/test_cli.py (hunk 2 above)
Comment (ffapitalle) [reply tests/test_cli.py]: Another typo (I'm not native English speaker also). I've changed it as you suggests, it's more clear.

File: yamllint/cli.py (hunk 4 above)
Comment (ffapitalle) [reply yamllint/cli.py]: I wanted to keep the value of the return code , even if I filter the output.

Added the extra test for checking both options given in the same run.

Comment (adrienverge) [reply yamllint/cli.py]: :+1:
